ktor-utils/common/src/io/ktor/util/date/GMTDateParser.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.date
+
+import io.ktor.util.*
+
+private typealias DateChunkParser = GMTDateBuilder.(String) -> Unit
+
+/**
+ * Build [GMTDate] parser using [pattern] string.
+ *
+ * Pattern string format:
+ * | Unit     | pattern char | Description                                          |
+ * | -------- | ------------ | ---------------------------------------------------- |
+ * | Seconds  | s            | parse seconds 0 to 60                                |
+ * | Minutes  | m            | parse minutes 0 to 60                                |
+ * | Hours    | h            | parse hours 0 to 23                                  |
+ * | Month    | M            | parse month from Jan to Dec(see [Month] for details) |
+ * | Year     | Y            | parse year                                           |
+ * | Any char | *            | Match any character                                  |
+ *
+ * All other chars are checked to match exact pattern. If you want to exact match any of pattern chars,
+ * you should escape it with `\`:
+ *
+ * @sample: pattern "\s ss" matches string "s 15"
+ */
+@KtorExperimentalAPI
+class GMTDateParser(private val pattern: String) {
```

Comment:
Not sure if it should be open since it is definitely not a general purpose date parser (at least because of missing variable length component support, am I right?)